components/TimeSeriesList.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div
+    class="card-body p-0"
+  >
+    <TimeSeries
+      v-for="(_, seriesId, index) in appStore.timeSeriesData"
+      :key="seriesId"
+      :all-charts="charts"
+      :series-id="seriesId"
+      :index="index"
+      :open="openedAccordions.includes(seriesId)"
+      :hide-tooltips="hideTooltips"
+      :chart-height-px="chartHeightPx"
+      :min-chart-height-px="minChartHeightPx"
+      @hide-all-tooltips="hideAllTooltips"
+      @show-all-tooltips="showAllTooltips"
+      @store-chart="storeChart"
+      @unstore-chart="unstoreChart"
+      @sync-tooltips-and-crosshairs="syncTooltipsAndCrosshairs(seriesId)"
+      @toggle-open="toggleOpen(seriesId)"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import throttle from "lodash.throttle";
+
+const appStore = useAppStore();
+
+const charts = ref<Record<string, Highcharts.Chart>>({});
+const openedAccordions = ref<string[]>([]);
+const hideTooltips = ref(false);
+const accordionBodyYPadding = 8;
+const minAccordionHeight = 150;
+const maxAccordionHeight = 400;
+const minTotalAccordionHeight = 500;
+const minChartHeightPx = minAccordionHeight - (2 * accordionBodyYPadding);
+const maxTotalAccordionHeight = computed(() => {
+  if (appStore.timeSeriesData) { // Allow at least minAccordionHeight for each accordion
+    return Math.max(minTotalAccordionHeight, (Object.keys(appStore.timeSeriesData!).length * minAccordionHeight));
+  } else {
+    return minTotalAccordionHeight;
+  }
+});
+// Share available height equally between open accordions. Avoid division by zero.
+const accordionHeight = computed(() => openedAccordions.value.length ? (maxTotalAccordionHeight.value / openedAccordions.value.length) : 1);
+const chartHeightPx = computed(() => Math.min(accordionHeight.value, maxAccordionHeight) - (2 * accordionBodyYPadding));
+
+const storeChart = (seriesId: string, chart: Highcharts.Chart) => {
```

Comment:
Because `charts` is fully managed in this parent component, and does not interact with the pinia store, I feel like `storeChart`/`unstoreChart` is potentially a bit misleading. How about `chartCreated`/`chartDestroyed` or `chartAdded`/`chartRemoved`?